File: app/src/main/java/com/moufee/purduemenus/util/AuthHelper.java
package com.moufee.purduemenus.util;

import okhttp3.FormBody;
import okhttp3.Request;

public class AuthHelper {

    public static final String favoritesURL = "https://api.hfs.purdue.edu/menus/v2/favorites";


    public static Request getTGTRequest(String username, String password) {
        FormBody formBody = new FormBody.Builder()
                .add("username", username)
                .add("password", password)
                .build();
        String ticketURL = "https://www.purdue.edu/apps/account/cas/v1/tickets";
        return new Request.Builder()
                .url(ticketURL)
                .post(formBody)
                .build();
    }

    public static Request getTicketRequest(String location, String service) {
        FormBody ticketRequestBody = new FormBody.Builder()
                .add("service", service)
                .build();

        return new Request.Builder()
                .url(location)
                .post(ticketRequestBody)
                .build();
    }
}

File: app/src/main/java/com/moufee/purduemenus/util/AuthHelper.kt
package com.moufee.purduemenus.util

import androidx.annotation.VisibleForTesting
import okhttp3.FormBody
import okhttp3.Request

object AuthHelper {
    @VisibleForTesting
    var apiBaseUrl = "https://api.hfs.purdue.edu"

    @VisibleForTesting
    var authBaseUrl = "https://www.purdue.edu"

    val favoritesURL get() = "$apiBaseUrl/menus/v2/favorites"
    private val ticketUrl get() = "$authBaseUrl/apps/account/cas/v1/tickets"

    fun getTGTRequest(username: String, password: String): Request {
        val formBody: FormBody = FormBody.Builder()
                .add("username", username)
                .add("password", password)
                .build()

        return Request.Builder()
                .url(ticketUrl)
                .post(formBody)
                .build()
    }

    fun getTicketRequest(location: String, service: String): Request {
        val ticketRequestBody: FormBody = FormBody.Builder()
                .add("service", service)
                .build()
        return Request.Builder()
                .url(location)
                .post(ticketRequestBody)
                .build()
    }
}